File: delivery-java-project/api/src/main/java/org/delivery/api/config/web/WebConfig.java
package org.delivery.api.config.web;

import java.util.List;

import org.delivery.api.interceptor.AuthorizationInterceptor;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.servlet.config.annotation.InterceptorRegistry;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;

import lombok.RequiredArgsConstructor;

@Configuration
@RequiredArgsConstructor
public class WebConfig implements WebMvcConfigurer {
  private final AuthorizationInterceptor authorizationInterceptor;
  private List<String> OPEN_API = List.of(
      "/open-api/**");

  private List<String> DEFAULT_EXCLUDE = List.of(
      "/", "/favicon.ico", "/error");

  private List<String> SWAGGER = List.of("/swagger-ui.html", "/swagger-ui/**", "/swagger-resources/**", "/v2/api-docs");

  @Override
  public void addInterceptors(InterceptorRegistry registry) {
    registry.addInterceptor(authorizationInterceptor)
        .excludePathPatterns(OPEN_API).excludePathPatterns(DEFAULT_EXCLUDE).excludePathPatterns(SWAGGER);
  }
}

File: delivery-java-project/api/src/main/java/org/delivery/api/domain/token/web/WebConfig.kt
package org.delivery.api.domain.token.web

import org.delivery.api.interceptor.AuthorizationInterceptor
import org.delivery.api.resolver.UserSessionResolver
import org.springframework.context.annotation.Configuration
import org.springframework.web.method.support.HandlerMethodArgumentResolver
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer

@Configuration
class WebConfig(
	private val authorizationInterceptor: AuthorizationInterceptor,
	private val userSessionResolver: UserSessionResolver
): WebMvcConfigurer {
	private val OPEN_API = listOf(
		"/open-api/**"
	)
	
	private val DEFAULT_EXCLUDE = listOf(
		"/", "/favicon.ico", "/error"
	)
	
	private val SWAGGER =
		listOf("/swagger-ui.html", "/swagger-ui/**", "/swagger-resources/**", "/v2/api-docs", "/v3/api-docs/**")
	
	override fun addArgumentResolvers(resolvers: MutableList<HandlerMethodArgumentResolver>) {
		resolvers.add(userSessionResolver)
	}
}